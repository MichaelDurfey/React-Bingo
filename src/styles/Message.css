:local(.message) {
  display: flex;
  position: absolute;
  top: calc((50% - 87px));
  flex-flow: column;
  justify-content: center;
  align-items: center;
  height: 20%;
  justify-content: center;
  width: 100%;
}

:local(.message > div) {
  border: solid 1px black;
  font-size: 25px;
  display: flex;
  justify-content: center;
  align-items: center;
  font-family: Arial, Helvetica, sans-serif;
  background-color: rgb(245, 247, 248);
  width: 50%;
  height: 100%;
}

:local(div.messageHidden) {
  display: none;
}

:local(div.messageShown) {
  animation-duration: 1s;
  animation-name: fadeIn;
}

@keyframes fadeIn { 
  from {
    opacity: 0;
  } to {
    opacity: 1;
  }
}