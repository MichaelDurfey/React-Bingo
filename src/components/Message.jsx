import React from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/index.css';

export default function Message(props) {
  const { message, className } = props;
  return (
    <div className={`${styles.message} ${className}`}>
      <div>
        { message }
      </div>
    </div>
  );
}

Message.propTypes = {
  message: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};
