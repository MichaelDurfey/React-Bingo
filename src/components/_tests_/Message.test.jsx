import React from 'react';
import renderer from 'react-test-renderer';
import Message from '../Message';

describe('Message', () => {
  test('it renders correctly with defaults', () => {
    const message = renderer
      .create(<Message message={''}/>)
      .toJSON();
    expect(message).toMatchSnapshot();
  });
  test('Message is shown depending on message existence', () => {
    const message = renderer
      .create(<Message message={'hi'} />)
      .toJSON();
    expect(message.props.className.includes('messageShown')).toBe(true);
  });
  test('Message is shown depending on message existence', () => {
    const message = renderer
      .create(<Message message={''} />)
      .toJSON();
    expect(message.props.className.includes('messageShown')).toBe(false);
    expect(message.props.className.includes('messageHidden')).toBe(true);
  });
});
