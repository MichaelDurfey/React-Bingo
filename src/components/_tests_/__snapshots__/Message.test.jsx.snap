// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message it renders correctly with defaults 1`] = `
<div
  className="message messageHidden"
>
  <div>
    
  </div>
</div>
`;
